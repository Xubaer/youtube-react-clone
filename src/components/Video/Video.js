import React from 'react';
import './Video.scss';

const BASED_EMBED_URL = 'https://www.youtube.com/embed/';

export function Video(props) {
    if(!props.id){
        return null;
    }
    const embedUrl = `${BASED_EMBED_URL}${props.id}`;
    return (
        <div className='video-container'>
            <div className="video">
                <iframe className='video-player' src={embedUrl} frameBorder='0' title={'Video'} allow='autoplay; encrypted-media' allowFullScreen />
            </div>
        </div>
    );
}