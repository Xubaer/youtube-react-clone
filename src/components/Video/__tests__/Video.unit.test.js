import React from 'react';
import { shallow } from 'enzyme';
import { Video } from '../Video';

describe('testing Video', () => {
    test('render without id', () => {
        const wrapper = shallow(
            <Video />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('renders video with id', () => {
        const wrapper = shallow(
            <Video id='123456' />
        );
        expect(wrapper).toMatchSnapshot();
    });
});