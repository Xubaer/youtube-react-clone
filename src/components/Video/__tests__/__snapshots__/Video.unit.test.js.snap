// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Video render without id 1`] = `""`;

exports[`testing Video renders video with id 1`] = `
<div
  className="video-container"
>
  <div
    className="video"
  >
    <iframe
      allow="autoplay; encrypted-media"
      allowFullScreen={true}
      className="video-player"
      frameBorder="0"
      src="https://www.youtube.com/embed/123456"
      title="Video"
    />
  </div>
</div>
`;
